src/Microsoft.AzureHealth.DataServices.Core/Clients/GenericRestClient.cs

```diff
@@ -0,0 +1,154 @@
+            this.logger = logger;
+        }
+
+        private readonly ILogger logger;
+        private readonly TokenCredential _tokenCredential;
+        private readonly HttpPipeline _pipeline;
+        private readonly Uri _endpoint;
+
+        /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+        public virtual HttpPipeline Pipeline => _pipeline;
+        /// <summary>
+        /// Default Content Type as Json
+        /// </summary>
+        private static string ContentType => "application/json";
+
+        /// <summary>
+        /// Sends and http request and returns a response.
+        /// </summary>
+        /// <returns>HttpResponseMessage</returns>
+        public async Task<Response> SendAsync(OperationContext operationContext)
+        {
+            try
+            {
+                //string token = await FetchToken();
+                using HttpMessage message = operationContext.Request.Method.ToString().ToUpperInvariant() switch
+                {
+                    "GET" => CreateRequest(operationContext),
+                    "POST" => CreateRequest(operationContext),
+                    "PUT" => CreateRequest(operationContext),
+                    "DELETE" => CreateRequest(operationContext),
+                    "PATCH" => CreateRequest(operationContext),
+                    _ => throw new Exception("Invalid Http method."),
+                };
+                await _pipeline.SendAsync(message, CancellationToken.None);
+                var response = message.Response;
+                logger?.LogInformation("Rest response returned status {StatusCode}.", response);
+                logger?.LogTrace("Rest response returned with content-type {ContentType}.", response?.Headers.ContentType);
+
+                if (response.Status == (int)HttpStatusCode.OK)
+                {
+                    logger?.LogInformation("Return http response.");
+                }
+                else
+                {
+                    logger?.LogWarning("Rest response returned fault reason phrase {ReasonPhrase}.", response.ReasonPhrase);
+                }
+
+                return response;
+            }
+            catch (WebException wex)
+            {
+                logger?.LogError(wex, "Rest web request faulted '{Status}'.", wex.Status);
+                throw;
+            }
+            catch (Exception ex)
+            {
+                logger?.LogError(ex, "Rest request faulted '{Message}'.", ex.Message);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Create Http Message Request.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        private HttpMessage CreateRequest(OperationContext context)
```

Comment:
Will this add any header in the context.Request.Headers collection? Like "x-ms-AuditId"?